File: 3872-find-most-frequent-vowel-and-consonant/3872-find-most-frequent-vowel-and-consonant.c
int maxFreqSum(char* s) {
    int i,j,n,f,c,g,sm;
    n=strlen(s);
    g=0;
    for(i=0;i<n;i++)
    if(s[i]=='a' ||s[i]=='e' ||s[i]=='i' ||s[i]=='o' ||s[i]=='u')
    {
        f=0;
        for(j=0;j<i;j++)
        if(s[i]==s[j])
        {
            f=1;
            break;
        }
        if(f==0)
        {
            c=0;
            for(j=i;j<n;j++)
            if(s[i]==s[j])
            c++;
        }
        if(c>g)
        g=c;
    }
        sm=0;
    for(i=0;i<n;i++)
    if(s[i]!='a' && s[i]!='e' && s[i]!='i' && s[i]!='o' && s[i]!='u')
    {
        f=0;
        for(j=0;j<i;j++)
        if(s[i]==s[j])
        {
            f=1;
            break;
        }
        if(f==0)
        {
            c=0;
            for(j=i;j<n;j++)
            if(s[i]==s[j])
            c++;
        }
        if(c>sm)
        sm=c;
    }
    return sm+g;
}

File: 3872-find-most-frequent-vowel-and-consonant/3872-find-most-frequent-vowel-and-consonant.java
class Solution {
    public int maxFreqSum(String s) {
        Map<Character,Integer> vowel = new HashMap<>();
        Map<Character,Integer> consonant = new HashMap<>();
        for(Character k : s.toCharArray())
        if(k == 'a' || k == 'e' || k == 'i' || k == 'o' || k == 'u')
        vowel.merge(k,1,Integer::sum);
        else
        consonant.merge(k,1,Integer::sum);
        int m1 = 0;
        if(vowel.values().stream().max((a,b) -> a.compareTo(b)).isPresent())
         m1 = vowel.values().stream().max((a,b) -> a.compareTo(b)).get();
        int m2 = 0;
        if(consonant.values().stream().max((a,b) -> a.compareTo(b)).isPresent())
         m2 = consonant.values().stream().max((a,b) -> a.compareTo(b)).get();
        return m1+m2;
    }

}